drivers/include/mcp23017.h

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2017 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_mcp23017
+ * @ingroup     drivers_actuators
+ * @brief       Device driver interface for the MCP23017
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the MCP23017
+ *
+ * @author      Dimitri Nahm <dimitri.nahm@haw-hamburg.de>
+ */
+
+#ifndef MCP23017_H
+#define MCP23017_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+/**
+ * @brief   Possible MCP23017 hardware addresses (wiring specific)
+ */
+enum {
+    MCP23017_ADDR_20 = 0x20, /**< I2C device address: A2=0, A1=0, A0=0 */
+    MCP23017_ADDR_21 = 0x21, /**< I2C device address: A2=0, A1=0, A0=1 */
+    MCP23017_ADDR_22 = 0x22, /**< I2C device address: A2=0, A1=1, A0=0 */
+    MCP23017_ADDR_23 = 0x23, /**< I2C device address: A2=0, A1=1, A0=1 */
+    MCP23017_ADDR_24 = 0x24, /**< I2C device address: A2=1, A1=0, A0=0 */
+    MCP23017_ADDR_25 = 0x25, /**< I2C device address: A2=1, A1=0, A0=1 */
+    MCP23017_ADDR_26 = 0x26, /**< I2C device address: A2=1, A1=1, A0=0 */
+    MCP23017_ADDR_27 = 0x27, /**< I2C device address: A2=1, A1=1, A0=1 */
+};
+
+/**
+ * @brief   Output Interrupt selection
+ */
+enum {
+    INTA,  /**< Output interrupt on INTA pin */
+    INTB   /**< Output interrupt on INTB pin */
+};
+
+/**
+ * @brief   Named return values
+ */
+enum {
+    MCP23017_OK          =  0,       /**< everything was fine */
+    MCP23017_NOI2C       = -1,       /**< I2C communication failed */
+    MCP23017_NODEV       = -2        /**< no MCP23017 device found on the bus */
+};
+
+/**
+ * @brief Device descriptor for the MCP23017 sensor
+ */
+typedef struct {
+    i2c_t i2c;                       /**< I2C device which is used */
+    uint8_t addr;                    /**< I2C address */
+    uint8_t port_a_dir;
+    uint8_t port_b_dir;
+    uint8_t port_a_pull_up;
+    uint8_t port_b_pull_up;
+    uint8_t port_a_value;
+    uint8_t port_b_value;
+    uint8_t port_a_inten;
+    uint8_t port_b_inten;
+} mcp23017_t;
+
+/**
+ * @brief   Struct containing the needed peripheral configuration
+ */
+typedef struct {
+    gpio_t int_a_pin;                /**< interrupt a pin */
+    gpio_t int_b_pin;                /**< interrupt b pin */
+} mcp23017_params_t;
```

Comment:
This struct is incomplete and inconsistent with the actual usage below. As I see it, it should contain:
```c
typedef struct {
    i2c_t i2c_dev;
    uint8_t i2c_addr;
    gpio_t int_a_pin;
    gpio_t int_b_pin;
} mcp23017_params_t;
```